### PR TITLE
MULE-16463: Remove `flatMap` in static <flow-ref>

### PR DESCRIPTION
**Pull request by elrodro83**
* Fallback to previous method when recursive flow-ref is found during
initialization

### CODE REVIEW

**Comment by plbalbi** [comment]
Great commit name

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/FlowRefFactoryBean.java [hunk 1]
```diff
@@ -230,8 +232,19 @@ protected StaticFlowRefMessageProcessor(FlowRefFactoryBean owner) {
...
+    protected StaticFlowRefMessageProcessor(FlowRefFactoryBean owner, DynamicFlowRefMessageProcessor recursiveFallback) {
+      super(owner);
+      this.recursiveFallback = recursiveFallback;
+    }
+
     @Override
     public Publisher<CoreEvent> apply(Publisher<CoreEvent> publisher) {
+      if (recursionFound) {
+        return from(publisher).transform(recursiveFallback);
+      }
```
Comment (plbalbi) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/FlowRefFactoryBean.java]: This is just to do the recursion check during the first apply right?

File: modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/FlowRefFactoryBean.java [hunk 2]
```diff
@@ -453,15 +482,14 @@ public Processor load(String key) throws Exception {
...
+     *
      * @param event the {@link CoreEvent} event
      * @return the {@link Processor} targeted by the current event
      * @throws MuleException
      */
     protected Processor resolveTargetFlowOrSubflow(CoreEvent event) throws MuleException {
-      String targetFlowName = (String) expressionManager.evaluate(refName, event, getLocation()).getValue();
       try {
-        return targetsCache.getUnchecked(targetFlowName);
+        return targetsCache.getUnchecked(refNameFromEvent.apply(event));
```
Comment (plbalbi) [inline modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/FlowRefFactoryBean.java]: Higher order ❤️

Comment (elrodro83) [reply modules/spring-config/src/main/java/org/mule/runtime/config/internal/factories/FlowRefFactoryBean.java]: ftw

**Pull request merged**